src/chocolatey.resources/helpers/functions/Install-BinFile.ps1

```diff
@@ -86,6 +86,13 @@ param(
 
   Write-FunctionCallLogMessage -Invocation $MyInvocation -Parameters $PSBoundParameters
 
+  if ($env:ChocolateyNoShims) {
+    Write-Debug "File shimming disabled for `'$($env:ChocolateyPackageName)`'."
+    Write-Debug "Removing any existing shim for `'$name`'."
+    Uninstall-BinFile $name $path
+    return
+  }

```

Comment:
Following my last comment, I see you have code to remove shims if no shims is called. Might be good to add a scenario for that use case

Reply:
Not quite sure what you want here, so I have added an extra upgrade scenario in https://github.com/chocolatey/choco/pull/2003/commits/b4963903d91cfa5f756ca1cb2aa584074ea6d426

This tests an upgrade to a package that has a shim, with the --noshims flag.

Is this the sort of thing you wanted?